lib/aytests/iso_repo.rb

```diff
@@ -50,7 +50,8 @@ def get(uri)
       end
       if download_to(uri)
         # Returning the first found ISO in this directory
-        Dir.entries(iso_dir).first
+        filename = Dir.entries(iso_dir).delete_if{ |f| File.directory?(f) || !f.end_with?(".iso") }.first
```

Comment:
unfortun. File.file?(f) is returning false every time  whereas File.directory?("..") is returning the correct value. We need the complete path for checking. So I am not sure if it looks clearer now :-)